### PR TITLE
http: Improving HTTP/1.1 proxy support

### PR DESCRIPTION
**Pull request by alyssawilk**
When the proxy filter was implemented it did direct string checks to ensure a 200 response.  This addresses a TODO to accept any well formed 200 response.

Risk Level: low
Testing: new unit tests, integration tests
Docs Changes: n/a
Release Notes: n/a
Part of https://github.com/envoyproxy/envoy-mobile/issues/1622

### CODE REVIEW

**Comment by alyssawilk** [comment]
cc @bencebeky I'm just straight up using balsa here if you're good with it.  It turns out http_parser stops one character early and I'd rather just use balsa.  This filter is being rolled out for Envoy Mobile and all this does is validate a 200 OK response for CONNECT requests

**Comment by bencebeky** [comment]
> cc @bencebeky I'm just straight up using balsa here if you're good with it. It turns out http_parser stops one character early and I'd rather just use balsa. This filter is being rolled out for Envoy Mobile and all this does is validate a 200 OK response for CONNECT requests

Sounds good to me.

**Comment by alyssawilk** [comment]
@snowp ping!

File: source/extensions/transport_sockets/http_11_proxy/connect.cc [hunk 1]
```diff
@@ -45,8 +58,8 @@ Network::IoResult UpstreamHttp11ConnectSocket::doWrite(Buffer::Instance& buffer,
 
 Network::IoResult UpstreamHttp11ConnectSocket::doRead(Buffer::Instance& buffer) {
   if (need_to_strip_connect_response_) {
-    // Limit the CONNECT response headers to an arbitrary 200 bytes.
-    constexpr uint32_t MAX_RESPONSE_HEADER_SIZE = 200;
+    // Limit the CONNECT response headers to an arbitrary 2000 bytes.
```
Comment (snowp) [inline source/extensions/transport_sockets/http_11_proxy/connect.cc:61]: I know this is arbitrary but why the increase?

**Pull request merged**